### PR TITLE
2070 add support for ttc concent enabled

### LINKED ISSUE #2070: Add support for the `TaskToCompute`'s `concent_enabled`
Add support for the `TaskToCompute`s `concent_enabled` flag where needed in Golem core logic

Related to https://github.com/golemfactory/golem-messages/issues/125

### PR DESCRIPTION
**Pull request by shadeofblue**
closes #2070 

+ refactor the tests slightly (dry-ify)

### CODE REVIEW

File: tests/golem/task/test_tasksession.py [hunk 1]
```diff
@@ -689,31 +689,24 @@ def setUp(self):
...
-        ts = TaskSession(mock.Mock())
-        n = Node()
-        task_id = str(uuid.uuid4())
-        subtask_id = str(uuid.uuid4())
-        node_id = str(uuid.uuid4())
-        wtr = WaitingTaskResult(task_id, subtask_id, "result", ResultType.DATA,
-                                13190, 10, 0, "10.10.10.10",
-                                30102, "key1", n, package_sha1="deadbeef")
-        task_to_compute = message.TaskToCompute()
+        del self.ts
```
Comment (mfranciszkiewicz) [inline tests/golem/task/test_tasksession.py]: These `del`s aren't really needed

**Review by mfranciszkiewicz — APPROVED** [review]
LGTM, though `pylint` has a complaint

File: golem/task/tasksession.py [hunk 2]
```diff
@@ -372,27 +372,28 @@ def send_report_computed_task(
...
-        # though, should we receive the acknowledgement for
-        # the `ReportComputedTask` sent above before the delay elapses,
-        # the `ForceReportComputedTask` message to the Concent will be
-        # cancelled and thus, never sent to the Concent.
-        msg = message.ForceReportComputedTask(
-            task_to_compute=task_to_compute,
-            result_hash='sha1:' + task_result.package_sha1
-        )
-        logger.debug('[CONCENT] ForceReport: %s', msg)
+        if task_to_compute.concent_enabled:
```
Comment (jiivan) [inline golem/task/tasksession.py]:
```python
if not task_to_compute.concent_enabled:
    return
```
?

File: tests/golem/task/test_tasksession.py [hunk 3]
```diff
@@ -726,62 +714,76 @@ def test_send_report_computed_task_concent_success(self):
         )
         service = history.MessageHistoryService.instance
         service.add_sync(nmsg_dict)
-        ts.send_report_computed_task(wtr, "10.10.10.10", 30102, "0x00", n)
-        ts.concent_service.submit.assert_called_once_with(
+
+    def _waiting_task_result(self, task_id, subtask_id, node,
+                             result='result', result_type=ResultType.DATA):  # noqa pylint:disable=too-many-arguments
```
Comment (jiivan) [inline tests/golem/task/test_tasksession.py]: Missing whitespace is why your silencer didn't work
`# noqa pylint: disable=too-many-arguments`

Comment (jiivan) [reply tests/golem/task/test_tasksession.py]: But I'm not sure if this should be silenced. Maybe it could be refactored by using partial?

**Review by jiivan — APPROVED** [review]

File: golem/task/tasksession.py (hunk 2 above)
Comment (shadeofblue) [reply golem/task/tasksession.py]: @jiivan right

**Pull request merged**